go-mysql panic when replicate mariadb

### What did you do?

start task with mariadb upstream

### What did you expect to see?

task runs well

### What did you see instead?

```
[2022/12/23 10:53:35.311 +08:00] [ERROR] [streamer_controller.go:360] ["meet error when get binlog event"] [task=test] [unit="binlog replication"] [error="Err: runtime error: invalid memory address or nil pointer dereference\n Stack: goroutine 528 [running]:\ngithub.com/go-mysql-org/go-mysql/mysql.Pstack(...)\n\t/home/gmhdbjd/environment/go/pkg/mod/github.com/go-mysql-org/go-mysql@v1.6.1-0.20221130104000-8206ccde58ed/mysql/util.go:22\ngithub.com/go-mysql-org/go-mysql/replication.(*BinlogSyncer).onStream.func1()\n\t/home/gmhdbjd/environment/go/pkg/mod/github.com/go-mysql-org/go-mysql@v1.6.1-0.20221130104000-8206ccde58ed/replication/binlogsyncer.go:694 +0x8d\npanic({0x69e6400, 0xac6ef50})\n\t/usr/local/go/src/runtime/panic.go:890 +0x262\ngithub.com/go-mysql-org/go-mysql/replication.(*BinlogSyncer).parseEvent(0xc0008cd520, 0xc001261300, {0xc000e50000, 0x2b, 0x30})\n\t/home/gmhdbjd/environment/go/pkg/mod/github.com/go-mysql-org/go-mysql@v1.6.1-0.20221130104000-8206ccde58ed/replication/binlogsyncer.go:832 +0x773\ngithub.com/go-mysql-org/go-mysql/replication.(*BinlogSyncer).onStream(0xc0008cd520, 0xc001261300)\n\t/home/gmhdbjd/environment/go/pkg/mod/github.com/go-mysql-org/go-mysql@v1.6.1-0.20221130104000-8206ccde58ed/replication/binlogsyncer.go:760 +0x63b\ncr"]
```

### Versions of the cluster

DM version (run `dmctl -V` or `dm-worker -V` or `dm-master -V`):
```console
(paste DM version here, and you must ensure versions of dmctl, DM-worker and DM-master are same)
```
Upstream MySQL/MariaDB server version:
```console
(paste upstream MySQL/MariaDB server version here)
```
Downstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):
```console
(paste TiDB cluster version here)
```
How did you deploy DM: tiup or manually?
```console
(leave TiUP or manually here)
```
Other interesting information (system version, hardware config, etc):
```console
>
>
```

### current status of DM cluster (execute `query-status <task-name>` in dmctl)

```console
(paste current status of DM cluster here)
```

--- a/go.mod
+++ b/go.mod
@@ -26,7 +26,7 @@ require (
 	github.com/gin-gonic/gin v1.7.4
 	github.com/glebarez/go-sqlite v1.17.3
 	github.com/glebarez/sqlite v1.4.6
-	github.com/go-mysql-org/go-mysql v1.6.1-0.20221130104000-8206ccde58ed
+	github.com/go-mysql-org/go-mysql v1.6.1-0.20221223014230-81966e15b9c5
 	github.com/go-ozzo/ozzo-validation/v4 v4.3.0
 	github.com/go-sql-driver/mysql v1.6.0
 	github.com/goccy/go-json v0.9.11

